polkadot-parachains/integritee-runtime/src/lib.rs

```diff
@@ -878,9 +879,21 @@ pub type NativeAndAssets = UnionOf<
 	NativeOrWithId<AssetIdForTrustBackedAssets>,
 	AccountId,
 >;
-pub type AscendingLocator = Ascending<AccountId, NativeOrWithId<AssetIdForTrustBackedAssets>>;
-pub type WithFirstAssetLocator =
-	WithFirstAsset<Native, AccountId, NativeOrWithId<AssetIdForTrustBackedAssets>>;
+
+pub type PoolIdToAccountId = pallet_asset_conversion::AccountIdConverter<
+	AssetConversionPalletId,
+	(NativeOrWithId<AssetIdForTrustBackedAssets>, NativeOrWithId<AssetIdForTrustBackedAssets>),
+>;
+
+pub type AscendingLocator =
+	Ascending<AccountId, NativeOrWithId<AssetIdForTrustBackedAssets>, PoolIdToAccountId>;
+
+pub type WithFirstAssetLocator = WithFirstAsset<
+	Native,
+	AccountId,
+	NativeOrWithId<AssetIdForTrustBackedAssets>,
+	PoolIdToAccountId,
+>;
```

Comment:
Todo: I just added these to make the code compile: we have to check if this is a sane config.

Reply:
will test the DEX flow like here: #279 & #285